core/unix/signal.c

```diff
@@ -3235,20 +3235,10 @@ thread_set_self_context(void *cxt, bool is_detach_external)
     ASSERT_NOT_IMPLEMENTED(false); /* PR 405694: can't use regular sigreturn! */
```

Comment:
> This adds a test and a fix for https://github.com/DynamoRIO/dynamorio/issues/6919.

Please add a line at the end of the description `Fixes: #6919`

(See https://dynamorio.org/page_code_reviews.html#sec_commit_messages)